Assets/Scripts/Model/Content/SecondEdition/Pilots/NabooRoyalN1Starfighter/PadmeAmidala.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Abilities.SecondEdition;
+using System.Collections;
+using System.Collections.Generic;
+using Upgrade;
+
+namespace Ship
+{
+  namespace SecondEdition.NabooRoyalN1Starfighter
+  {
+    public class PadmeAmidala : NabooRoyalN1Starfighter
+    {
+      public PadmeAmidala() : base()
+      {
+        PilotInfo = new PilotCardInfo(
+            "Padmé Amidala",
+            4,
+            45,
+            isLimited: true,
+            abilityText: "While an enemy ship in your [Front Arc] defends or performs an attack, that ship can modify only 1 [Focus] result (other results can still be modified).",
+            abilityType: typeof(PadmeAmidalaAbility),
+            extraUpgradeIcon: UpgradeType.Talent
+        );
+
+        ImageUrl = "https://images-cdn.fantasyflightgames.com/filer_public/80/40/8040bcab-ebc0-487e-8dff-bd69da7311dd/swz40_padme-amidala.png";
+      }
+    }
+  }
+}
+
+namespace Abilities.SecondEdition
+{
+  public class PadmeAmidalaAbility : GenericAbility
+  {
+    public override void ActivateAbility()
+    {
+      HostShip.AfterGotNumberOfPrimaryWeaponAttackDice += CheckAttackAbility;
```

Comment:
The activate and deactivate ability functions would need to be updated with an appropriate hook (likely before modifications begin and possibly also after each modification?)